src/main/kotlin/org/rust/ide/annotator/RsCargoCheckAnnotator.kt

```diff
@@ -147,18 +147,23 @@ private fun filterMessage(file: PsiFile, document: Document, message: RustcMessa
     val spanFilePath = PathUtil.toSystemIndependentName(span.file_name)
     if (!file.virtualFile.path.endsWith(spanFilePath)) return null
 
+    @Suppress("NAME_SHADOWING")
     fun toOffset(line: Int, column: Int): Int? {
+        val line = line - 1
+        val column = column - 1
```

Comment:
Well, my opinion seems to be totally opposite :)

I am currently working in Elixir and I don't remember any situation where I used shadowing except in `with` clause where pipe operator doesn't work:

```elixir
with \
  {:ok, foo} <- some_function(...),
  {:ok, foo} <- some_other_function(foo, ...)
do
    ...
end
```

What I don't like about such shadowing is, it is in fact *pure mutability* and can introduce nasty bugs when you forget that there is a shadow somewhere in a function.